frontend/src/routes/(app)/report/[owner]/[report]/+page.svelte

```diff
@@ -127,31 +120,31 @@
 					options={projects}
 				/>
 			{/await}
+			<hr class="mt-4 bg-grey-dark" />
 		{/if}
-		<hr class="mt-6 mb-2 text-grey-light" />
 		<div
-			class="flex flex-col"
-			use:dndzone={{ items: elements, dragDisabled: !dragEnabled }}
+			class="flex flex-col mt-2"
+			use:dndzone={{
+				items: elements,
+				dragDisabled: !dragEnabled,
+				dropTargetStyle: {},
+				flipDurationMs: 0
+			}}
 			on:consider={handleMoved}
 			on:finalize={handleDropped}
 		>
 			{#each elements as element (element.id)}
-				{#if isEdit}
-					<ElementEdit
-						bind:element
-						bind:dragEnabled
-						{chartOptions}
-						{sliceOptions}
-						reportId={data.report.id}
-						on:delete={() => deleteElement(element.id ?? -1)}
-					/>
-				{:else}
-					<Element {element} {chartOptions} />
-				{/if}
+				<ElementContainer
+					bind:element
+					bind:editId
+					bind:dragEnabled
+					bind:showConfirmDelete
+					{addElement}
+					{chartOptions}
+					{sliceOptions}
+					report={data.report}
+				/>
 			{/each}
-			{#if isEdit}
-				<Button variant="raised" on:click={() => addElement(elements.length)}>Add Element</Button>
-			{/if}
```

Comment:
I think there is no way for me to add an element if there is no element in the report already now.

Reply:
LOL good catch